dockstore-common/src/main/java/io/dockstore/common/yaml/DockstoreYamlHelper.java

```diff
@@ -373,40 +378,92 @@ private static void addNewPropertyClassToQueue(String dockstoreYamlPackageName,
                     // Sort them lexicographically by property path (ex "workflows[0].author[0].name").
                     // The result doesn't match their order in the yaml file, but is probably good enough for now...
                     .sorted((a, b) -> a.getPropertyPath().toString().compareTo(b.getPropertyPath().toString()))
-                    .map(v -> buildMessageFromViolation(v))  // NOSONAR here, a lambda is more understandable than method reference
+                    .map(v -> buildMessageFromViolation(v, targetDescription))  // NOSONAR a lambda is more understandable than method reference here
                     .collect(Collectors.joining("; ")));
         }
     }
 
-    private static <T> String buildMessageFromViolation(ConstraintViolation<T> violation) {
-        String message = violation.getMessage();
-
-        // If the violation contains a non-empty property path, add it to the message, with a summary of the invalid value, if present.
+    /**
+     * Creates an error message from a constraint violation, using a description of the
+     * target (object being validated) if the property is not available.
+     * @param violation constrain violation to convert
+     * @param targetDescription string that described the object being validated (ex: ".dockstore.yml")
+     * @return a well-formed error message
+     */
+    private static <T> String buildMessageFromViolation(ConstraintViolation<T> violation, String targetDescription) {
+        // Determine the subject of the error message, either a property name or a description of the target.
+        String subject = null;
         javax.validation.Path propertyPath = violation.getPropertyPath();
         if (propertyPath != null) {
             String propertyPathString = propertyPath.toString();
             if (propertyPathString != null && !propertyPathString.isEmpty()) {
-                message = String.format("Property \"%s\" %s", propertyPathString, message);
-                Object invalidValue = violation.getInvalidValue();
-                if (invalidValue != null) {
-                    message = String.format("%s (current value: \"%s\")", message, StringUtils.abbreviate(invalidValue.toString(), INVALID_VALUE_ECHO_LIMIT));
-                }
+                subject = String.format("Property \"%s\"", propertyPathString);
             }
         }
+        if (subject == null) {
+            subject = targetDescription;
+        }
+
+        // Create the message by prepending the subject to the verb phrase in the violation message.
+        String message = String.format("%s %s", subject, violation.getMessage());
+
+        // If there's a non-null invalid value and its toString() method overrides the base Object.toString(), include its string representation in the message.
+        Object invalidValue = violation.getInvalidValue();
+        if (invalidValue != null && !invalidValue.toString().equals(objectToString(invalidValue))) {
```

Comment:
I made a small improvement here, the message now includes the string representation of the invalid value if the invalid value is not null and the output of its `toString()` is different than the default (the output of `Object.toString()`).